app/controllers/registrations_controller.rb

```diff
@@ -22,6 +22,7 @@ def create
       end
     else
       clean_up_passwords @user
+      flash[:notice] = flash[:notice].to_a.concat(resource.errors.full_messages).join(", ")
```

Comment:
Can you check edge cases for this? I get an error sometimes when I hit submit multiple times in a row.